grails-app/controllers/au/org/ala/layers/LayerController.groovy

```diff
@@ -353,44 +354,48 @@ class LayerController {
 
     }
 
+    @RequireAdmin
     def download(String id) {
         Layer l = layerDao.getLayerByDisplayName(id)
-        if (downloadAllowed(l)) {
-            OutputStream outputStream = null
-            try {
-                outputStream = response.outputStream as OutputStream
-                //write resource
-                response.setContentType("application/octet-stream")
-                response.setHeader("Content-disposition", "attachment;filename=${id}.zip")
-
-                // When a geotiff exists, only download the geotiff
-                def path = "/layer/${l.name}"
-                def geotiff = new File(grailsApplication.config.data.dir + path + ".tif")
-                if (geotiff.exists()) {
-                    path += ".tif"
-                }
+        if(l){
+            if (downloadAllowed()) {
+                OutputStream outputStream = null
+                try {
+                    outputStream = response.outputStream as OutputStream
+                    //write resource
+                    response.setContentType("application/octet-stream")
+                    response.setHeader("Content-disposition", "attachment;filename=${id}.zip")
+
+                    // When a geotiff exists, only download the geotiff
+                    def path = "/layer/${l.name}"
+                    def geotiff = new File(grailsApplication.config.data.dir + path + ".tif")
+                    if (geotiff.exists()) {
+                        path += ".tif"
+                    }
 
-                fileService.write(outputStream, path)
-                outputStream.flush()
-            } catch (err) {
-                log.error(err.getMessage(), err)
-            } finally {
-                if (outputStream != null) {
-                    try {
-                        outputStream.close()
-                    } catch (err) {
-                        log.error(err.getMessage(), err)
+                    fileService.write(outputStream, path)
+                    outputStream.flush()
+                } catch (err) {
+                    log.error(err.getMessage(), err)
+                } finally {
+                    if (outputStream != null) {
+                        try {
+                            outputStream.close()
+                        } catch (err) {
+                            log.error(err.getMessage(), err)
+                        }
                     }
                 }
+            } else {
+                response.sendError(403, "Downloding $id is prohabited by licence!")
             }
-        } else {
+        }else{
             response.sendError(404, "$id not available")
         }
     }
 
     private downloadAllowed(layer) {
-        return grailsApplication.config.download.layer.licence_levels.contains(layer.licence_level) ||
-                serviceAuthService.isAdmin(params)
+        return grailsApplication.config.download.layer.licence_levels.contains(layer.licence_level)
```

Comment:
You will get NPE if one of the config vars is not set. Can be avoided by using Grails utility method:
`grailsApplication.config.getProperty('download.layer.licence_levels', String, '')`.